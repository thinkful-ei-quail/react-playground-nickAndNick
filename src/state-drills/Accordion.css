/* Here will be our beautiful CSS */