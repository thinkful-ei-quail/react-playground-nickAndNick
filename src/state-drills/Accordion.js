import React, { Component } from 'react';

class Accordion extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isToggleOn: true
        }
    }

    renderButton() {
        if (!this.state.isToggleOn) {
            return (
                <li>
                    <button onClick={this.expandSectionClick}>
                        Dummy Title
                        {this.props.sections.title}
                    </button>
                </li>
            )
        } else {
            return (
                <li>
                    <button onClick={this.expandSectionClick}>
                        Dummy Title
                        {this.props.sections.title}
                        <p>Dummy Content</p>
                        <p>{this.props.sections.content}</p>
                    </button>
                </li>
            )
        } 
    }

    expandSectionClick = () => {
        this.setState({
            isToggleOn: !this.state.isToggleOn
        })
        // console.log(this.state.isToggleOn)
        this.renderButton();
    }

    render() {
        return (
            <ul>
                <li>
                    <button onClick={this.expandSectionClick} >
                        Dummy Title
                        {this.props.sections.title}
                    </button>
                </li>
            </ul>
        )
    }
}




export default Accordion;