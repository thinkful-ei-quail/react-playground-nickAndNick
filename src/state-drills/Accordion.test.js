import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import './Accordion.css';
import renderer from 'react-test-renderer';

describe(`Accordion Component`, () => {
  it('renders without errors', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion sections={[{title:'Section 1', content:'Lorem ipsum dolor sit amet consectetur adipisicing elit.'}]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  })

  it('renders the UI as expected', () => {
    const tree = renderer
    .create(<Accordion sections={[{title:'Section 1', content:'Lorem ipsum dolor sit amet consectetur adipisicing elit.'}]}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();  
  });

})