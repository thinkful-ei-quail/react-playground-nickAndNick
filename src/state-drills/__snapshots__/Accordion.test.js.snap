// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component opens a clicked section 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders an empty li when sections prop is not supplied 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders no sections active as default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
</ul>
`;

exports[`Accordion Component renders the UI as expected 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
</ul>
`;
